File: examples/text_to_video.py
from __future__ import annotations

import os
from pprint import pprint

from seedance25_api import Seedance25Client


def main() -> None:
    api_key = os.environ["CYBERBARA_API_KEY"]
    client = Seedance25Client(api_key)

    quote = client.quote_video(
        prompt="A cinematic drone shot over a misty mountain village at sunrise.",
        scene="text-to-video",
        options={"duration": "10", "aspect_ratio": "16:9"},
    )
    pprint({"quote": quote})

    created = client.text_to_video(
        "A cinematic drone shot over a misty mountain village at sunrise.",
        duration="10",
        aspect_ratio="16:9",
    )
    pprint({"created": created})

    task = client.wait_for_task(created["task_id"])
    pprint({"task": task, "videos": task.get("output", {}).get("videos", [])})


if __name__ == "__main__":
    main()

File: examples/text_to_video.mjs
import { Seedance25Client } from "../index.js";

const client = new Seedance25Client(process.env.CYBERBARA_API_KEY);

const quote = await client.quoteVideo({
  prompt: "A cinematic drone shot over a misty mountain village at sunrise.",
  scene: "text-to-video",
  options: {
    duration: "10",
    aspect_ratio: "16:9"
  }
});

console.log({ quote });

const created = await client.textToVideo(
  "A cinematic drone shot over a misty mountain village at sunrise.",
  {
    duration: "10",
    aspectRatio: "16:9"
  }
);

console.log({ created });

const task = await client.waitForTask(created.task_id);
console.log({ task, videos: task.output?.videos || [] });
